### PR TITLE
[cleanup] Misc

### PR DESCRIPTION
**Pull request by Grub4K**
Closes #8248, Closes #11146, Closes #11149, Closes #11211

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is released under [Unlicense](http://unlicense.org/): Authored by: @grqz, @DTrombett, @KarboniteKream, @bashonly, @mikkovedru, @seproDev 

### What is the purpose of your *pull request*?
- [x] Docs/Cleanup
</details>

### CODE REVIEW

File: yt_dlp/YoutubeDL.py [hunk 1]
```diff
@@ -4085,17 +4084,6 @@ def get_encoding(stream):
         if plugin_dirs:
             write_debug(f'Plugin directories: {plugin_dirs}')
 
-        # Not implemented
```
Comment (grqz) [inline yt_dlp/YoutubeDL.py:4088]: maybe also add the deprecation warning back?
```diff
diff --git a/yt_dlp/YoutubeDL.py b/yt_dlp/YoutubeDL.py
index b67b68886..5676a01c0 100644
--- a/yt_dlp/YoutubeDL.py
+++ b/yt_dlp/YoutubeDL.py
@@ -551,7 +551,7 @@ class YoutubeDL:
                        (requires writesubtitles or writeautomaticsub)
     include_ads:       - Doesn't work
                        Download ads as well
-    call_home:         - Not implemented
+    call_home:         - Deprecated
                        Boolean, true iff we are allowed to contact the
                        yt-dlp servers for debugging.
     post_hooks:        - Register a custom postprocessor
diff --git a/yt_dlp/__init__.py b/yt_dlp/__init__.py
index f598b6c2f..f88b49efa 100644
--- a/yt_dlp/__init__.py
+++ b/yt_dlp/__init__.py
@@ -556,8 +556,8 @@ def report_deprecation(val, old, new=None):
 
     report_deprecation(opts.sponskrub, '--sponskrub', '--sponsorblock-mark or --sponsorblock-remove')
     report_deprecation(not opts.prefer_ffmpeg, '--prefer-avconv', 'ffmpeg')
+    report_deprecation(opts.call_home, '--call-home')
     # report_deprecation(opts.include_ads, '--include-ads')  # We may re-implement this in future
-    # report_deprecation(opts.call_home, '--call-home')  # We may re-implement this in future
     # report_deprecation(opts.writeannotations, '--write-annotations')  # It's just that no website has it
 
     # Dependent options
```

Comment (bashonly) [reply yt_dlp/YoutubeDL.py:4088]: nah, the CLI opts need to remain so that youtube-dl configs can be ported to yt-dlp

Comment (bashonly) [reply yt_dlp/YoutubeDL.py:4088]: this is also forgetting about `--no-call-home`

**Comment by grqz** [comment]
Some other cleanup prs that can be included here: #11207, #11149, #11146, #10853, #8248

**Comment by bashonly** [comment]
> Some other cleanup prs that can be included here: https://github.com/yt-dlp/yt-dlp/pull/11207, https://github.com/yt-dlp/yt-dlp/pull/11149, https://github.com/yt-dlp/yt-dlp/pull/11146, https://github.com/yt-dlp/yt-dlp/pull/10853, https://github.com/yt-dlp/yt-dlp/pull/8248

#11207: I prefer to have extractor removals in their own commits for visibility 

#11149: Done

#11146: Done

#10853: Not convinced it's necessary

#8248: Labeled `do-not-merge`

**Comment by grqz** [comment]
For #8248 it's labeled as `do-not-merge` because the author changed the README without changing options.py, which will be overwritten in the next release. options.py was updated in https://github.com/yt-dlp/yt-dlp/pull/8248/commits/6248e526cf681257ff4ff5062665df6e3a858c1f and I think the label can be removed

File: yt_dlp/extractor/veoh.py [hunk 2]
```diff
@@ -108,7 +109,7 @@ def _real_extract(self, url):
 
         categories = metadata.get('categoryPath')
         if not categories:
-            category = try_get(video, lambda x: x['category'].strip().removeprefix('category_'))
+            category = remove_start(strip_or_none(video.get('category')), 'category_')
             categories = [category] if category else None
         tags = video.get('tags')
```
Comment (grqz) [inline yt_dlp/extractor/veoh.py:115]: this has already been included in Grub4K#1, would you like to review it?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -268,7 +268,7 @@ After you have ensured this site is distributing its content legally, you can fo
 
     You can use `hatch fmt` to automatically fix problems. Rules that the linter/formatter enforces should not be disabled with `# noqa` unless a maintainer requests it. The only exception allowed is for old/printf-style string formatting in GraphQL query templates (use `# noqa: UP031`).
 
-1. Make sure your code works under all [Python](https://www.python.org/) versions supported by yt-dlp, namely CPython and PyPy for Python 3.8 and above. Backward compatibility is not required for even older versions of Python.
+1. Make sure your code works under all [Python](https://www.python.org/) versions supported by yt-dlp, namely CPython >=3.8 and PyPy >=3.10. Backward compatibility is not required for even older versions of Python.
```
Comment (gamer191) [inline CONTRIBUTING.md:271]: Should this say 3.9, in preparation for removing 3.8?

File: README.md [hunk 4]
```diff
@@ -173,11 +173,11 @@ python3 -m pip install -U --pre "yt-dlp[default]"
 ```
 
 ## DEPENDENCIES
-Python versions 3.8+ (CPython and PyPy) are supported. Other versions and implementations may or may not work correctly.
+Python versions 3.8+ (CPython) and 3.10+ (PyPy) are supported. Other versions and implementations may or may not work correctly.
```
Comment (gamer191) [inline README.md:176]: Should this say 3.9?

File: pyproject.toml [hunk 5]
```diff
@@ -172,13 +173,11 @@ run-cov = "echo Code coverage not implemented && exit 1"
 
 [[tool.hatch.envs.hatch-test.matrix]]
 python = [
-    "3.8",
     "3.9",
     "3.10",
     "3.11",
     "3.12",
-    "pypy3.8",
-    "pypy3.9",
```
Comment (gamer191) [inline pyproject.toml:181]: Why did you remove Pypy3.9?

File: CONTRIBUTING.md (hunk 3 above)
Comment (bashonly) [reply CONTRIBUTING.md:271]: we'll do that in the upcoming "remove Python 3.9 support" PR

File: pyproject.toml (hunk 5 above)
Comment (bashonly) [reply pyproject.toml:181]: It's dead. PyPy now only supports 3.10 (and 2.7)

**Comment by grqz** [comment]
in the docstring of InfoExtractor:
RSE102: [Unnecessary parentheses on raised exception](https://docs.astral.sh/ruff/rules/unnecessary-paren-on-raise-exception/)

<details><summary>fix</summary>

```diff
diff --git a/yt_dlp/extractor/common.py b/yt_dlp/extractor/common.py
index d55f34238..795105b7d 100644
--- a/yt_dlp/extractor/common.py
+++ b/yt_dlp/extractor/common.py
@@ -520,7 +520,7 @@ class InfoExtractor:
     or _extract_from_webpage as necessary. While these are normally classmethods,
     _extract_from_webpage is allowed to be an instance method.
 
-    _extract_from_webpage may raise self.StopExtraction() to stop further
+    _extract_from_webpage may raise self.StopExtraction to stop further
     processing of the webpage and obtain exclusive rights to it. This is useful
     when the extractor cannot reliably be matched using just the URL,
     e.g. invidious/peertube instances
```

</details>

**Comment by grqz** [comment]
<details>
<summary>
patch:

remove `title` as a mandatory field of infodict in CONTRIBUTING.md
</summary>

```diff
diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
index 968c69f88..99e9d91ae 100644
--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -302,10 +302,9 @@ ### Mandatory and optional metafields
 For extraction to work yt-dlp relies on metadata your extractor extracts and provides to yt-dlp expressed by an [information dictionary](yt_dlp/extractor/common.py#L119-L440) or simply *info dict*. Only the following meta fields in the *info dict* are considered mandatory for a successful extraction process by yt-dlp:
 
  - `id` (media identifier)
- - `title` (media title)
  - `url` (media download URL) or `formats`
 
-The aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken. While all extractors must return a `title`, they must also allow it's extraction to be non-fatal.
+The aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken. All extractors must allow it's extraction to be non-fatal.
 
 For pornographic sites, appropriate `age_limit` must also be returned.
 
```

</details>

Also, it seems like some links to `common.py` are broken, but I'm confused about where these links are supposed to go to

https://github.com/yt-dlp/yt-dlp/blob/fbc66e3ab35743cc847a21223c67d88bb463cd9c/CONTRIBUTING.md?plain=1#L302
for the `information dictionary` link, maybe this is just part of it?

https://github.com/yt-dlp/yt-dlp/blob/fbc66e3ab35743cc847a21223c67d88bb463cd9c/CONTRIBUTING.md?plain=1#L314
This `Any field` link is definitely broken

**Comment by Grub4K** [comment]
I just noticed while investigating your other hints that inside `common.py` theres another:
https://github.com/yt-dlp/yt-dlp/blob/d710a6ca7c622705c0c8c8a3615916f531137d5d/yt_dlp/extractor/common.py#L120-L125

> Also, it seems like some links to common.py are broken, but I'm confused about where these links are supposed to go to
> This `Any field` link is definitely broken

Both link to the docstring inside `common.py` which explains the individual fields. I think its hard to get these sort of links right. If anything gets added or removed or shifted around these will not match up. Maybe best to link to the file in general and just mention the info being in the docstring?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -302,10 +302,9 @@ Extractors are very fragile by nature since they depend on the layout of the sou
 For extraction to work yt-dlp relies on metadata your extractor extracts and provides to yt-dlp expressed by an [information dictionary](yt_dlp/extractor/common.py#L119-L440) or simply *info dict*. Only the following meta fields in the *info dict* are considered mandatory for a successful extraction process by yt-dlp:
 
  - `id` (media identifier)
- - `title` (media title)
  - `url` (media download URL) or `formats`
 
-The aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken. While all extractors must return a `title`, they must also allow it's extraction to be non-fatal.
+The aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken. All extractors must make all other metadata extraction non-fatal.
```
Comment (bashonly) [inline CONTRIBUTING.md:307]: let's clean this up a bit more; the first sentence is kind rambly
```suggestion
The aforementioned metadata fields are the critical data without which extraction does not make any sense. If any of them fail to be extracted, then the extractor is considered broken. All other metadata extraction should be completely non-fatal.
```
